src/main/scala-sbt-0.13/compat.scala

```diff
@@ -89,6 +89,12 @@ package internal {
     object `package`
   }
 
+  package util {
+    object `package` {
+      val JLine: { def usingTerminal[T](f: jline.Terminal => T): T } = sbt.JLine
```

Comment:
Is the type annotation necessary?

Reply:
Removing it entirely produces the error `private object JLine escapes its defining scope as part of type sbt.JLine.type`

Reply:
Ah right. In that case I think we should just implement this as a JLine object in the util package, instead of a val in the util package object. That way we avoid using reflection:

```scala
Welcome to Scala 2.12.3 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_144).
Type in expressions for evaluation. Or try :help.

scala> object Foo { object Bar { def baz: Int = 1 }; val Quux: { def baz: Int } = Bar }
defined object Foo

scala> Foo.Quux.baz
<console>:13: warning: reflective access of structural type member method baz should be enabled
by making the implicit value scala.language.reflectiveCalls visible.
This can be achieved by adding the import clause 'import scala.language.reflectiveCalls'
or by setting the compiler option -language:reflectiveCalls.
See the Scaladoc for value scala.language.reflectiveCalls for a discussion
why the feature should be explicitly enabled.
       Foo.Quux.baz
                ^
res0: Int = 1
```